src/verify/verify.test.ts

```diff
@@ -309,3 +309,169 @@ describe("isValidVc", () => {
+      expect.objectContaining({
+        body: JSON.stringify({
+          verifiablePresentation: MOCK_VP,
+          options: { domain: "domain", challenge: "challenge" },
+        }),
+      })
+    );
+  });
+
+  it("uses the provided verification endpoint if any", async () => {
+    const mockedFetch = jest.fn(global.fetch).mockResolvedValueOnce(
+      new Response(JSON.stringify(MOCK_VERIFY_RESPONSE), {
+        status: 200,
+      })
+    );
+    mocked(isVerifiablePresentation).mockReturnValueOnce(true);
+
+    await isValidVerifiablePresentation(
+      "https://some.verification.api",
+      MOCK_VP,
+      {
+        fetch: mockedFetch,
+      }
+    );
+    expect(mockedFetch).toHaveBeenCalledWith(
+      "https://some.verification.api",
+      expect.anything()
+    );
+    expect(getVerifiableCredentialApiConfiguration).not.toHaveBeenCalled();
+  });
+
+  it("discovers the verification endpoint if none is provided", async () => {
+    const mockedDiscovery = mocked(
+      getVerifiableCredentialApiConfiguration
+    ).mockResolvedValueOnce({
+      verifierService: "https://some.vc.verifier",
+      legacy: {},
+      specCompliant: {},
+    });
+    mocked(isVerifiablePresentation).mockReturnValueOnce(true);
+
+    const mockedFetch = jest.requireMock("../fetcher") as jest.Mocked<{
+      default: typeof fetch;
+    }>;
+    mockedFetch.default.mockResolvedValueOnce(
+      new Response(JSON.stringify(MOCK_VERIFY_RESPONSE), {
+        status: 200,
+      })
+    );
+
+    await isValidVerifiablePresentation(null, MOCK_VP);
+    expect(mockedDiscovery).toHaveBeenCalledWith(MOCK_VP.holder);
+  });
+
+  it("throws if no verification endpoint is discovered", async () => {
+    mocked(getVerifiableCredentialApiConfiguration).mockResolvedValueOnce({
+      legacy: {},
+      specCompliant: {},
+    });
+    mocked(isVerifiablePresentation).mockReturnValueOnce(true);
+    const mockedFetch = jest.fn(global.fetch).mockResolvedValueOnce(
+      new Response(JSON.stringify(MOCK_VERIFY_RESPONSE), {
+        status: 200,
+      })
+    );
+
+    await expect(
+      isValidVerifiablePresentation(null, MOCK_VP, {
+        fetch: mockedFetch,
+      })
+    ).rejects.toThrow(
+      `The VC service provider ${MOCK_VP.holder} does not advertize for a verifier service in its .well-known/vc-configuration document`
+    );
+  });
+
+  it("throws if the verification endpoint returns an error", async () => {
+    const mockedFetch = jest.fn(global.fetch).mockResolvedValueOnce(
+      new Response(undefined, {
+        status: 400,
+        statusText: "Bad request",
+      })
+    );
+    mocked(isVerifiablePresentation).mockReturnValueOnce(true);
+
+    await expect(
+      isValidVerifiablePresentation(MOCK_VERIFY_ENDPOINT, MOCK_VP, {
+        fetch: mockedFetch as typeof fetch,
+      })
+    ).rejects.toThrow(/consent\.example\.com.*400 Bad request/);
+  });
+
+  it("returns the validation result from the access endpoint", async () => {
```

Comment:
why for this case is it called "access endpoint" vs verification endpoint? both use MOCK_VERIFY_ENDPOINT so I'm just making sure I understand.

Reply:
ah good catch! updated.